include/optimizer.h

```diff
@@ -24,8 +24,9 @@ class Optimizer {
   /*!
   * \brief Operator constructor, the optimizer is not initialized until the first Update
   * \param opt_type type of the optimizer
+  * \param learning_rate
   */
-  explicit Optimizer(const std::string &opt_type);
+  Optimizer(const std::string &opt_type, mx_float learning_rate);
```

Comment:
In distributed setting, it's server's responsibility to update parameters with optimizer, and the [prototype](https://github.com/dmlc/MXNet.cpp/blob/master/include/c_api.h#L1071) of updater determines that the optimizer should maintain internally all the parameters including lr and wd. 
Current python [implementation](https://github.com/dmlc/mxnet/blob/master/python/mxnet/optimizer.py#L276) use the same strategy. If the worker does need to control lr or wd, I can add setters for them in optimizer.